### PR TITLE
Multiple Price different move from anglo saxon module

### PR DESCRIPTION
**Pull request by bizzappdev**
Upstream PR: https://github.com/odoo/odoo/pull/704
## Bug In case of Same product appears more than one time in single supplier invoice. 

**Possibility are when / Step to reproduce.**
- Pre-Configuration required in category/product Price different account, Stock valuation Account, Stock Input Account, Stock Output Account. account_anglo_saxon module is installed.
- We have purchase order with unit price different than Cost price and QTY 500.
- Invoicing Control = Based on incoming shipments
- We have 5 lots when receive products from supplier So Split lines into 5 lines.
- Create invoice from the Incoming Shipment.
- It will create Supplier invoice with 5 Lines with same product .
- Validate Supplier Invoice.
  It will create 5 move line for product account with cost price, Tax line (if any), Total creditor account, and 5*5 = 25 Price different move lines. 

**Here the bug is it has to have only 5 price different move lines.**

Cause is in  account_anglo_saxon/invoice.py it is comparing move lines (provided by move_line_get method ) and invoice line with account_id and product_id.
So for all 5 lines account_id and product_id are same so it is creating lines n times for each invoice line.

**Changes done.**
- Add Line Id from move line get in account/account_invoice.py
- Add condition while generating price difference move account_anglo_saxon/invoice.py.

### CODE REVIEW

File: addons/account/account_invoice.py [hunk 1]
```diff
@@ -133,7 +133,7 @@ def _amount_residual(self, cr, uid, ids, name, args, context=None):
                     result[invoice.id] = new_value
 
             #prevent the residual amount on the invoice to be less than 0
-            result[invoice.id] = max(result[invoice.id], 0.0)            
+            result[invoice.id] = max(result[invoice.id], 0.0)
```
Comment (StefanRijnhart) [inline addons/account/account_invoice.py]: Please avoid unnecessary whitespace changes such as these

File: addons/account_anglo_saxon/invoice.py [hunk 2]
```diff
@@ -115,6 +115,8 @@ def get_price(cr, uid, inv, company_currency,i_line):
                         diff_res = []
                         # calculate and write down the possible price difference between invoice price and product price
                         for line in res:
+                            if line.get('invl_id', False) != i_line.id:
```
Comment (StefanRijnhart) [inline addons/account_anglo_saxon/invoice.py]: You can compare line.get('invl_id') to i_line.id because of get()'s default None value. Or is there a chance that i_line is a browse_null record?

File: addons/account/account_invoice.py (hunk 1 above)
Comment (bizzappdev) [inline addons/account/account_invoice.py]: Don't you think it is always better not to have extra white spaces ?

File: addons/account_anglo_saxon/invoice.py (hunk 2 above)
Comment (bizzappdev) [inline addons/account_anglo_saxon/invoice.py]: No there are no chances of browse_null record. Will change.

File: addons/account/account_invoice.py (hunk 1 above)
Comment (StefanRijnhart) [inline addons/account/account_invoice.py]: Looks like these changes are making the code more pep-8 compliant, but whitespace changes in other lines than which are affected by the fix increase the chances of conflicts with upstream Odoo on the long run.

Comment (bizzappdev) [inline addons/account/account_invoice.py]: Yes, You are perfectly right,  But I Have provided Same PR also for the Odoo https://github.com/odoo/odoo/pull/704. And If they will also accept it then There is not problem for upstream conflicts .

Comment (StefanRijnhart) [inline addons/account/account_invoice.py]: If they accept it you don't need to propose on OCB

Comment (StefanRijnhart) [inline addons/account/account_invoice.py]: Note that we're on GitHub now. Only one branch to update your pull requests on OCB ánd on upsteam Odoo!

Comment (bizzappdev) [inline addons/account/account_invoice.py]: Yes Only one branch is proposed to both . that is right.
But you mean to say . if it will be merged in odoo then it will automatically merged in OCB ?

Comment (StefanRijnhart) [inline addons/account/account_invoice.py]: No, that is not the case but when an OCB specific change is merged in Odoo, the change will automatically be exactly the same.

Comment (bizzappdev) [inline addons/account/account_invoice.py]: ahh then this is other case . My branch is from odoo 7.0 . and making PR for both . I think this is right way .

Comment (StefanRijnhart) [inline addons/account/account_invoice.py]: So, can we expect you to revert the changes in whitespace?

Comment (bizzappdev) [inline addons/account/account_invoice.py]: Sure in 30 mins...

**Comment by bizzappdev** [comment]
remove whitespace changes are done .

**Comment by pedrobaeza** [comment]
:+1:

**Comment by hbrunn** [comment]
:+1:

**Comment by bizzappdev** [comment]
merged in Odoo so closing this PR
